src/index.ts

```diff
@@ -253,7 +271,7 @@ async function stopProcessIfUnsuccessfulResults(): Promise<void> {
     const diagrams = [];
     RESULTS.forEach((res, scenario) => {
         diagrams.push(generateSequenceDiagram(scenario));
-        if (res.some(t => t.successful === false)) {
+        if (res.some(t => t.successful === false && t.allowFailure === false)) {
```

Comment:
I know it has been like this before the change but comparing against boolean values should be avoided. The line should be
```js
         if (res.some(t => !t.successful && !t.allowFailure)) {
```
instead.

Reply:
as discussed